### PR TITLE
various typos + typos configuration + CI

### PR DESCRIPTION
**Pull request by phip1611**
This fixes all typos I've found using the <https://github.com/crate-ci/typos> utility.
Furthermore, it adds an additional step to execute `typos` in GitHub CI.

### CODE REVIEW

File: .typos.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+[default.extend-words]
+# External Interrupt
+EXTINT = "EXTINT"
+extint = "extint"
+INOUT = "INOUT"
+Ded = "Ded" # GUIDed
+ba = "ba"
+Cros = "Cros" # CrosVM
+Dake = "Dake"
+Cachable = "Cachable"
```
Comment (rbradford) [inline .typos.toml:18]: https://github.com/rust-vmm/acpi_tables/issues/20

File: .typos.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "resources/linux-config-*",
+    "hypervisor/src/kvm/x86_64/mod.rs",
+]
+
+[default.extend-words]
+# External Interrupt
+EXTINT = "EXTINT"
+extint = "extint"
+INOUT = "INOUT"
+Ded = "Ded" # GUIDed
```
Comment (rbradford) [inline .typos.toml:14]: I reckon that is just a typo.

File: .typos.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ba = "ba"
+Cros = "Cros" # CrosVM
+Dake = "Dake"
+Cachable = "Cachable"
+outout = "outout"
+CLASSE = "CLASSE"
+conectix = "conectix"
+Requestor = "Requestor"
```
Comment (rbradford) [inline .typos.toml:22]: en_GB vs en_US but I think we should standardise on the *er* version

File: .typos.toml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+[default.extend-words]
+# External Interrupt
+EXTINT = "EXTINT"
+extint = "extint"
+INOUT = "INOUT"
+Ded = "Ded" # GUIDed
+ba = "ba"
+Cros = "Cros" # CrosVM
+Dake = "Dake"
```
Comment (rbradford) [inline .typos.toml:17]: I wonder what's special about that name and not any of the others?

**Review by rbradford — COMMENTED** [review]
Cool!

File: .typos.toml (hunk 4 above)
Comment (phip1611) [reply .typos.toml:17]: image

**Comment by phip1611** [comment]
Hi @rbradford ,
feel free to re-review: https://github.com/cloud-hypervisor/cloud-hypervisor/compare/2e2c029b86573d9026800354d9581b4152de05c9..06d8854eb7d7c0665db50ff51d96ec37f6fe6f12

File: .typos.toml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    "hypervisor/src/kvm/x86_64/mod.rs",
+]
+
+[default.extend-words]
+# External Interrupt
+EXTINT = "EXTINT"
+extint = "extint"
+INOUT = "INOUT"
+ba = "ba"
+Cros = "Cros" # CrosVM
```
Comment (alyssais) [inline .typos.toml:15]: The official [crosvm](https://crosvm.dev/book/running_crosvm/index.html) documentation seems to consistently spell it "crosvm", all lowercase, so maybe we should follow suit and not have this exception?

Comment (phip1611) [reply .typos.toml:15]: The typos utility is a little funny sometimes. It only complains about "Cros" in "CrossVM" and doesn't recognize the world as a whole.

File: .typos.toml (hunk 4 above)
Comment (iggy) [reply .typos.toml:15]: You would think the tool would be able to identify that pretty common email format (`Name Name <email>`) and not flag it

Comment (phip1611) [reply .typos.toml:15]: Agree, unfortunately it does not.

File: .typos.toml (hunk 1 above)
Comment (likebreath) [reply .typos.toml:17]: Are we holding this PR for the pending issue from acpi_tables crate?

Comment (rbradford) [reply .typos.toml:17]: This is fixed now so if you can rebase your version and update.

Comment (phip1611) [reply .typos.toml:17]: Done

**Review by likebreath — COMMENTED** [review]
Thank you for the contributions. Generally look good to me. Couple of cosmetic comments about the commit style. Please refer our contribution doc for complete details [1].

1. Please merge the last commit about crosvm to the fist commit;
2. Please rename the first commit to: `misc: Fix various typos`;
3. Please capitalize the first world for each commit subject after the component name.

[1] https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/CONTRIBUTING.md

**Comment by phip1611** [comment]
> Thank you for the contributions. Generally look good to me. Couple of cosmetic comments about the commit style. Please refer our contribution doc for complete details [1].
> 
> 1. Please merge the last commit about crosvm to the fist commit;
> 2. Please rename the first commit to: `misc: Fix various typos`;
> 3. Please capitalize the first world for each commit subject after the component name.
> 
> [1] https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/CONTRIBUTING.md

I updated the MR. I hope, we are good to go now. If not, let me know.

**Review by likebreath — APPROVED** [review]

**Comment by rbradford** [comment]
Please rebase.

**Comment by phip1611** [comment]
> Please rebase.

Done.

**Pull request merged**